src/main/scala/PropertyChecker.scala

```diff
@@ -402,4 +412,84 @@ object PropertyChecker {
+    def tableSchemaPropertyInternal(value: JsonNode, baseUrl: String, lang: String): (JsonNode, Array[String], PropertyType.Value) = {
+      var schemaBaseUrl = new URL(baseUrl)
+      var schemaLang = lang
+
+      var schemaJson: ObjectNode = null
+      if(value.isTextual) {
+        val schemaUrl = new URL(new URL(baseUrl), value.asText())
+        val objectMapper = new ObjectMapper()
+        schemaJson = objectMapper.readTree(schemaUrl).asInstanceOf[ObjectNode]
+        if (!schemaJson.path("@id").isMissingNode) {
+          // Do something here as object node put method doesn't allow uri object
+          val absoluteSchemaUrl = new URL(schemaUrl, schemaJson.get("@id").asText())
+          schemaJson.put("@id", absoluteSchemaUrl.toString)
+        } else {
+          schemaJson.put("@id", schemaUrl.toString)
+        }
+        val (newSchemaBaseUrl, newSchemaLang) = fetchSchemaBaseUrlAndLangAndRemoveContext(schemaJson, schemaBaseUrl, schemaLang)
+        schemaBaseUrl = newSchemaBaseUrl
+        schemaLang = newSchemaLang
+      } else if(value.isObject) {
+        schemaJson = value.deepCopy()
+      } else {
+        return (NullNode.getInstance(), Array[String]("invalid_value"), PropertyType.Table)
+      }
+
+      var warnings = Array[String]()
+      val fieldsAndValues = Array.from(schemaJson.fields.asScala)
+      for(fieldAndValue <- fieldsAndValues) {
+        val p = fieldAndValue.getKey
+        val v = fieldAndValue.getValue
+        if(p == "@id") {
+          val matcher = PropertyChecker.startsWithUnderscore.pattern.matcher(v.asText())
+          if (matcher.matches) {
+            throw new MetadataError(s"@id ${v.asText} starts with _:")
+          }
+        } else if(p == "@type") {
+          if (v.asText() != "Schema") {
+            throw new MetadataError("@type of schema is not 'Schema'")
+          }
+        } else {
+          val (v1, warningsForP, typeString) = checkProperty(p, v, schemaBaseUrl.toString, schemaLang)
+          if((typeString == PropertyType.Schema || typeString == PropertyType.Inherited) && warningsForP.isEmpty) {
+            schemaJson.set(p, v1)
+          } else {
+            schemaJson.remove(p)
+            if(typeString != PropertyType.Schema ||typeString !=PropertyType.Inherited) {
+              warnings = warnings :+ "invalid_property"
+            }
+          }
+        }
+      }
+      return (schemaJson, warnings, PropertyType.Table)
+    }
+    return tableSchemaPropertyInternal
+  }
+  def fetchSchemaBaseUrlAndLangAndRemoveContext(schemaJson:ObjectNode, schemaBaseUrl: URL, schemaLang:String): (URL, String) = {
+    if(!schemaJson.path("@context").isMissingNode) {
+      if (schemaJson.isArray && schemaJson.size > 1) {
+        val elements = Array.from(schemaJson.get("@context").elements.asScala)
+        val maybeBaseNode = elements.apply(1).path("@base")
```

Comment:
Two calls to `elements.apply(1)`, best if you rationalise them into one variable.

Reply:
Done